src/plugins/intel_gna/frontend/model_quantizer.hpp

```diff
@@ -22,125 +22,125 @@
+        return quantize(model, [](const InferenceEngine::CNNNetwork&, bool run_before_copy, bool low_precision){}, scale_factor);
     }
 
     template <class PreQuantisationCb>
-    InferenceEngine::CNNNetwork quantize(const InferenceEngine::CNNNetwork &model, const PreQuantisationCb &cb, float scaleFactor) const {
-        return quantize(model, cb, std::vector<float>({scaleFactor}));
+    InferenceEngine::CNNNetwork quantize(const InferenceEngine::CNNNetwork& model, const PreQuantisationCb& cb, float scale_factor) const {
+        return quantize(model, cb, std::vector<float>({scale_factor}));
     }
 
-    InferenceEngine::CNNNetwork quantize(const InferenceEngine::CNNNetwork &model, std::vector<float> scaleFactors) const {
-        return quantize(model, [](InferenceEngine::CNNNetwork &, bool runBeforeCopy, bool lowPrecision){}, scaleFactors);
+    InferenceEngine::CNNNetwork quantize(const InferenceEngine::CNNNetwork& model, std::vector<float> scale_factors) const {
+        return quantize(model, [](InferenceEngine::CNNNetwork&, bool run_before_copy, bool low_precision){}, scale_factors);
     }
 
     template <class PreQuantisationCb>
-    InferenceEngine::CNNNetwork quantize(const InferenceEngine::CNNNetwork &model,  const PreQuantisationCb &cb, std::vector<float> scaleFactors) const {
+    InferenceEngine::CNNNetwork quantize(const InferenceEngine::CNNNetwork& model,  const PreQuantisationCb& cb, std::vector<float> scale_factors) const {
         GNAPluginNS::GnaInputs inputs;
         InferenceEngine::InputsDataMap inputs_map = model.getInputsInfo();
         int sf_id = 0;
         for (auto &&input_data : inputs_map) {
             auto input_layer = getCreatorLayer(input_data.second->getInputData()).lock();
-            if (scaleFactors.size() <= sf_id) {
+            if (scale_factors.size() <= sf_id) {
                 THROW_GNA_EXCEPTION << "Scale factors are not set for some of the inputs";
             }
-            inputs[input_layer->name].scale_factor = scaleFactors[sf_id++];
+            inputs[input_layer->name].scale_factor = scale_factors[sf_id++];
         }
 
-        return quantize(model, cb, inputs);
+        return quantize(model, cb, inputs, InferenceEngine::Precision::I16, false);
     }
 
+    // This function is acutally used by the plugin
     template <class PreQuantisationCb>
-    InferenceEngine::CNNNetwork quantize(const InferenceEngine::CNNNetwork &model,
-                                         const PreQuantisationCb &cb,
-                                         const GNAPluginNS::GnaInputs &inputs) const {
+    InferenceEngine::CNNNetwork quantize(const InferenceEngine::CNNNetwork& model,
+                                         const PreQuantisationCb& cb,
+                                         const GNAPluginNS::GnaInputs& inputs,
+                                         const InferenceEngine::Precision& gna_precision,
```

Comment:
It is better to make these variables members if ModelQuantizer class if we decided do not use templates. Now arguments of this function are unclear.

Reply:
IMO the previous version of the code was unclear - you weren't able to tell what precision would be used for inputs, weights, biases. Now two new parameters are passed explicitly to show what precision configuration will be used. The template was only adding complexity and blocking flexibility, because it tied together weights and biases precisions.